optuna/study/study.py

```diff
@@ -54,7 +54,12 @@
 
 
 class _ThreadLocalStudyAttribute(threading.local):
-    in_optimize_loop: bool = False
```

Comment:
Is there something disadvantage to use the default value here? It looks like a common pattern as used in the following pep.
https://peps.python.org/pep-0567/#converting-code-that-uses-threading-local

Reply:
not really. will use default value implementation here.